File: hybride.py

class A:
    def funcA(self):
        print("This is the class A.")

class D:
    def funcD(self):
        print("This is the Class D.")

class B(A):
    def funcB(self):
        print("This is the class B.")

class C(A, D):
    def funcC(self):
        print("this is the class C.")


# Note In python class always be callable.

a = A()
a.funcA()

b = B()
b.funcA()
b.funcB()

c = C()
c.funcA()
c.funcC()
c.funcD()

d = D()
d.funcD()

File: hybride.cpp
#include<bits/stdc++.h>
using namespace std;

class A{
    public:
        void funcA(){
            cout<<"This is class A"<<endl;
        }
};

class D{
    public:
        void funcD(){
            cout<<"This is class D"<<endl;
        }
};

class B: public A{
    public:
        void funcB(){
            cout<<"This is Class B"<<endl;
        }
};

class C: public A, public D{
    public:
        void funcC(){
            cout<<"This is class C"<<endl;
        }
};

int main(){

    A a;
    a.funcA();

    B b;
    b.funcB();

    C c;
    c.funcA();
    c.funcC();
    c.funcD();

    D d;
    d.funcD();

    
    return 0;
}